shared/rangeanalysis/codeql/rangeanalysis/RangeAnalysis.qll

```diff
@@ -160,17 +160,62 @@ signature module Semantic {
   /** Gets a tiebreaker id in case `getBlockId1` is not unique. */
   default string getBlockId2(BasicBlock bb) { result = "" }
 
+  /**
+   * A guard in the range analysis.
+   */
   class Guard {
+    /**
+     * Gets a string representation of the guard.
+     */
     string toString();
 
+    /**
+     * Gets the basic block associated with the guard.
+     */
     BasicBlock getBasicBlock();
 
+    /**
+     * Gets the guard as an expression.
+     */
     Expr asExpr();
 
+    /**
+     * Holds if the guard directly controls a given basic block. For example in
+     * the following code, the guard `(x > y)` directly controls the block
+     * beneath it:
+     * ```
+     * if (x > y)
+     * {
+     *   Console.WriteLine("x is greater than y");
+     * }
+     * ```
+     *
+     * @param controlled The basic block to check.
+     */
     predicate directlyControls(BasicBlock controlled, boolean branch);
 
+    /**
+     * Holds if the guard represents an equality between two expressions.
+     *
+     * @param e1 The first expression.
```

Comment:
This is not the first instance of the problem, but why is there no documentation for `e2` and `polarity`?

Reply:
Yeah, and `polarity` is the interesting argument here that really deserved explanation...

Reply:
Addressed.  Thanks for pointing this out.